butterknife-compiler/src/test/java/butterknife/RClassTest.java

```diff
@@ -0,0 +1,234 @@
+      + "  }\n"
+      + "  public static final class bool {\n"
+      + "    public static final int res = 0x7f040002;\n"
+      + "  }\n"
+      + "  public static final class color {\n"
+      + "    public static final int res = 0x7f040003;\n"
+      + "  }\n"
+      + "  public static final class integer {\n"
+      + "    public static final int res = 0x7f040004;\n"
+      + "  }\n"
+      + "  public static final class string {\n"
+      + "    public static final int res = 0x7f040005;\n"
+      + "  }\n"
+      + "  public static final class styleable {\n"
+      + "    public static final int[] ActionBar = { 0x7f010001, 0x7f010003 };\n"
+      + "  }\n"
+      + "}"
+  );
+
+  private static final JavaFileObject R2 = JavaFileObjects.forSourceString("test.R2", ""
+      + "package test;\n"
+      + "public final class R2 {\n"
+      + "  public static final class array {\n"
+      + "    public static final int res = 0x7f040001;\n"
+      + "  }\n"
+      + "  public static final class bool {\n"
+      + "    public static final int res = 0x7f040002;\n"
+      + "  }\n"
+      + "  public static final class color {\n"
+      + "    public static final int res = 0x7f040003;\n"
+      + "  }\n"
+      + "  public static final class integer {\n"
+      + "    public static final int res = 0x7f040004;\n"
+      + "  }\n"
+      + "  public static final class string {\n"
+      + "    public static final int res = 0x7f040005;\n"
+      + "  }\n"
+      + "}"
+  );
+
+  @Test public void library() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.Test", ""
+        + "package test;\n"
+        + "import android.app.Activity;\n"
+        + "import butterknife.BindInt;\n"
+        + "import butterknife.RClass;\n"
+        + "@RClass(R.class)\n"
+        + "public class Test extends Activity {\n"
+        + "  @BindInt(R2.integer.res) int one;\n"
+        + "}"
+    );
+
+    JavaFileObject expectedSource = JavaFileObjects.forSourceString("test/Test$$ViewBinder", ""
+        + "package test;\n"
+        + "import android.content.res.Resources;\n"
+        + "import butterknife.Unbinder;\n"
+        + "import butterknife.internal.Finder;\n"
+        + "import butterknife.internal.ViewBinder;\n"
+        + "import java.lang.Object;\n"
+        + "import java.lang.Override;\n"
+        + "import java.lang.SuppressWarnings;\n"
+        + "public class Test$$ViewBinder<T extends Test> implements ViewBinder<T> {\n"
+        + "  @Override\n"
+        + "  public Unbinder bind(Finder finder, T target, Object source) {\n"
+        + "    Resources res = finder.getContext(source).getResources();\n"
+        + "    bindToTarget(target, res);\n"
+        + "    return Unbinder.EMPTY;\n"
+        + "  }\n"
+        + "  @SuppressWarnings(\"ResourceType\")\n"
+        + "  protected static void bindToTarget(Test target, Resources res) {\n"
+        + "    target.one = res.getInteger(test.R.integer.res);\n"
```

Comment:
Looks like we can clean this up so that it's not always going to use a FQCN. I can do that in a follow-up.